collections/vectors/vector.go

```diff
@@ -65,7 +69,17 @@ func (vector *Vector) ReadAll() (data []byte) {
 }
 
 func (vector *Vector) Consume(size int) (data []byte) {
-	// Todo: implement this
+	data = vector.body[:size]
+	vector.body = vector.body[size:]
```

Comment:
I think it's better to allocate a new body with make()

Reply:
because this will most likely break. vector body's true size after this call can be actually different than vector.size